src/libcore/str/mod.rs

```diff
@@ -1416,11 +1455,12 @@ impl StrExt for str {
     }
 
     #[inline]
-    fn splitn<'a, P: Pattern<'a>>(&'a self, count: usize, pat: P) -> SplitN<'a, P> {
+    fn splitn<'a, P: Pattern<'a>>(&'a self, count: usize, pat: P) -> SplitN<'a, P>
+        where P::Searcher: Searcher<'a>
```

Comment:
I think that the associated type on the `Pattern` trait already has this bound, so shouldn't this extra clause be unnecessary?